compiler/rustc_trait_selection/src/traits/codegen.rs

```diff
@@ -35,8 +35,12 @@ pub fn codegen_fulfill_obligation<'tcx>(
         let mut selcx = SelectionContext::new(&infcx);
 
         let obligation_cause = ObligationCause::dummy();
-        let obligation =
-            Obligation::new(obligation_cause, param_env, trait_ref.to_poly_trait_predicate());
+        let predicate = ty::Binder::dummy(ty::TraitPredicate {
```

Comment:
I think we should still be passing in a PolyTraitRef. Rather than using Binder::dummy here.

Reply:
hmm, not sure if we will ever end up calling `codegen_select_candidate` with something higherranked, but changed it back ^^ it doesn't hurt